uwblueprint/focus-on-nature
# Camp Overview: Remove Camper(s) from session

## Notion ticket link

[Camp Overview: Remove Camper(s) from session](https://www.notion.so/uwblueprintexecs/Camp-Overview-Remove-Camper-s-from-session-9c2c0522fc6c41f9a3e34203f2d0bd65)



## Implementation description
* Created new functions to delete campers by id and get campers that have the same specified charge id and session id, in `CamperAPIClient.ts`
* Added implementation for `getCampersByChargeIdAndSessionId` in the `CamperService.ts` files
* Created a new route for this endpoint to use
* Made the modals to interact with everything
![image](https://user-images.githubusercontent.com/30396273/195475310-59110798-9a7e-4496-b4f6-1f3810d220ce.png)
![image](https://user-images.githubusercontent.com/30396273/195475323-89aeed05-dcbd-42a5-a371-251f54d0585d.png)



## Steps to test
1. Navigate to the Guelph camp for instance, http://localhost:3000/admin/camp/63139c7bc3d7b55b44a01531
2. Remove different campers, ensure it works



## What should reviewers focus on?
* Ensure that removing campers with different session ids and charge ids are seperated (i.e. in the checkbox that pops up when removing, the checkbox should only contain campers with the same charge and session ids; if a camper doesn't share their session or charge id with anyone else, the select modal should not pop up)


## Checklist
